src/output_ir.rs

```diff
@@ -45,12 +57,13 @@ pub struct ScheduledNode {
 /// An [InsertedDelay] represents a required delay node to be inserted
 /// along some edge in order to compensate for different latencies along
 /// paths of the graph.
-#[derive(Copy, Clone, Debug, Serialize, Deserialize)]
+#[cfg_attr(feature = "serialize", derive(Serialize, Deserialize))]
+#[derive(Copy, Clone, Debug)]
 pub struct InsertedDelay {
     /// The edge that this delay corresponds to. Kept for debugging and visualization.
     pub edge: Edge,
     /// The amount of delay to apply to the input.
-    pub delay: f64,
+    pub delay: u64,
```

Comment:
Delay may be negative and non-integer.

Reply:
Delay in audio graphs is always in units of samples, there is no use for fractional delay.

Does negative delay actually work in this algorithm? If so, that's awesome! I can change it to an `i64` in that case.

Reply:
The units depends on the caller, it may be in milliseconds (fractional samples are also possible, like in lookahead). 

Converting from f64 to i64 keeps 52 bits of precision which should be plenty.